server/application-server/src/main/java/de/tum/in/www1/hephaestus/leaderboard/SlackMessageService.java

```diff
@@ -0,0 +1,165 @@
+
+@Order(value = Ordered.LOWEST_PRECEDENCE)
+@EnableScheduling
+@Service
+public class SlackMessageService {
+
+    private static final Logger logger = LoggerFactory.getLogger(SlackMessageService.class);
+
+    @Value("${slack.channelId}")
+    private String channelId;
+
+    @Value("${slack.runScheduledMessage}")
+    private boolean runScheduledMessage;
+
+    @Value("${spring.url:localhost:8080}")
+    private String hephaestusUrl;
+
+    @Autowired
+    private App slackApp;
+
+    @Autowired
+    private LeaderboardService leaderboardService;
+
+    public void sendMessage(String channelId, List<LayoutBlock> blocks, String fallback)
+            throws IOException, SlackApiException {
+        ChatPostMessageRequest request = ChatPostMessageRequest.builder()
+                .channel(channelId)
+                .blocks(blocks)
+                .text(fallback) // used for example in notifications
+                .build();
+
+        ChatPostMessageResponse response = slackApp.client().chatPostMessage(request);
+
+        if (!response.isOk()) {
+            logger.error("Failed to send message to Slack channel: " + response.getError());
+        }
+    }
+
+    private List<User> getTop3SlackReviewers() {
+        // exactly 7 days ago
+        OffsetDateTime after = OffsetDateTime.of(LocalDate.now().minusDays(7), OffsetDateTime.now().toLocalTime(),
+                OffsetDateTime.now().getOffset());
+        var leaderboard = leaderboardService.createLeaderboard(Optional.of(after), Optional.empty(),
+                Optional.empty());
+        var top3 = leaderboard.subList(0, Math.min(3, leaderboard.size()));
+        logger.debug("Top 3 Users of the last week: " + top3.stream().map(e -> e.user().name()).toList());
+
+        List<User> allSlackUsers;
+        try {
+            allSlackUsers = slackApp.client().usersList(r -> r).getMembers();
+        } catch (SlackApiException | IOException e) {
+            logger.error("Failed to get Slack users: " + e.getMessage());
+            return new ArrayList<>();
+        }
+
+        return top3.stream().map(entry -> {
+            var exactUser = allSlackUsers.stream()
+                .filter(user -> user.getName().equals(entry.user().name()) || (user.getProfile().getEmail() != null && user.getProfile().getEmail().equals(entry.user().email()))).findFirst();
+            if (exactUser.isPresent()) {
+                return exactUser.get();
+            }
+
+            // find through String edit distance
+            return allSlackUsers.stream()
+                .min((a, b) -> Integer.compare(
+                    LevenshteinDistance.getDefaultInstance().apply(
+                        entry.user().name(), a.getName()),
+                    LevenshteinDistance.getDefaultInstance().apply(
+                        entry.user().name(), b.getName()))).orElse(null);
+        }).filter(user -> user != null).toList();
+    }
+
+    @Scheduled(cron = "${slack.cronSchedule}")
```

Comment:
As commented above we could have it like this:
 - `hephaestus.leaderboard.weekly.reset.day`, e.g. `2` for Tuesday
 - `hephaestus.leaderboard.weekly.reset.time`, e.g. `9:00` for 9 A.M.

And then calculate the scheduled interval from that, otherwise we might misconfigure it

Reply:
Maybe I misunderstood your previous comments, but what would be the advantage over a CRON expression?
We've been using CRONs for other tasks (e.g. Github sync scheduler) for quite some time without issues now.

Reply:
I mean, we only support weekly leaderboards and have a reset week date and time, also this configuration has to match. Just to avoid some errors and to restrict the configuration options I would derive it somehow from the weekly notification settings

Reply:
As discussed on Slack, this is now done through a custom `ThreadPoolTaskScheduler`, which dynamically creates a new scheduler with a CRON expression derived from the env-values for `day` and `time`.
Probably way over-the-top, but should be pretty clean :)